vendor/github.com/openshift/cluster-api-actuator-pkg/pkg/e2e/machinehealthcheck/machinehealthcheck.go

```diff
@@ -0,0 +1,225 @@
+package machinehealthcheck
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+
+	"github.com/golang/glog"
+	e2e "github.com/openshift/cluster-api-actuator-pkg/pkg/e2e/framework"
+	mapiv1beta1 "github.com/openshift/cluster-api/pkg/apis/machine/v1beta1"
+	healthcheckingv1alpha1 "github.com/openshift/machine-api-operator/pkg/apis/healthchecking/v1alpha1"
+	"github.com/openshift/machine-api-operator/pkg/util/conditions"
+
+	corev1 "k8s.io/api/core/v1"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/types"
+
+	runtimeclient "sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+var _ = Describe("[Feature:MachineHealthCheck] MachineHealthCheck controller", func() {
+	var client runtimeclient.Client
+	var numberOfReadyWorkers int
+	var workerNode *corev1.Node
+	var workerMachine *mapiv1beta1.Machine
+
+	stopKubeletAndValidateMachineDeletion := func(workerNode *corev1.Node, workerMachine *mapiv1beta1.Machine, timeout time.Duration) {
+		By(fmt.Sprintf("Stopping kubelet service on the node %s", workerNode.Name))
+		err := e2e.StopKubelet(workerNode.Labels["kubernetes.io/hostname"])
```

Comment:
Node hostname does not always has to be equal to node name. Plus, you can use `workerNode.Name` directly.

Reply:
under the pod creation I use
```
NodeSelector: map[string]string{
      "kubernetes.io/hostname": nodeName,
},
```
so it for sure will use the right node

Reply:
You can set `.Spec.NodeName` directly to `nodeName`.